cmd/bcloud/config/config.go

```diff
@@ -1,37 +1,114 @@
 package config
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 
+	v1 "github.com/brevdev/cloud/pkg/v1"
 	"gopkg.in/yaml.v3"
 )
 
-type Config struct {
-	Credentials map[string]CredentialConfig `yaml:"credentials"`
-	Settings    Settings                    `yaml:"settings"`
+var providerRegistry = map[string]func() v1.CloudCredential{}
+
+func RegisterProvider(id string, factory func() v1.CloudCredential) {
+	providerRegistry[id] = factory
 }
 
-type CredentialConfig struct {
-	Provider          string `yaml:"provider"`
-	RefID             string `yaml:"ref_id,omitempty"`
-	APIKey            string `yaml:"api_key,omitempty"`
-	ServiceAccountKey string `yaml:"service_account_key,omitempty"`
-	ProjectID         string `yaml:"project_id,omitempty"`
-	DefaultLocation   string `yaml:"default_location,omitempty"`
+type CredentialEntry struct {
+	Provider string             `json:"provider" yaml:"provider"`
+	Value    v1.CloudCredential `json:"-" yaml:"-"`
 }
 
-type Settings struct {
-	OutputFormat   string `yaml:"output_format"`
-	DefaultTimeout string `yaml:"default_timeout"`
+func (c *CredentialEntry) decodeFromMap(m map[string]any, yamlKey string) error {
+	rawProv, ok := m["provider"]
+	if !ok {
+		return fmt.Errorf("missing 'provider'")
+	}
+	provider, ok := rawProv.(string)
+	if !ok || provider == "" {
+		return fmt.Errorf("invalid 'provider'")
+	}
+	factory, ok := providerRegistry[provider]
+	if !ok {
+		return fmt.Errorf("unknown provider: %s", provider)
+	}
+	cred := factory()
+
+	if _, hasRefID := m["ref_id"]; !hasRefID {
```

Comment:
lets write a test that says all registered providers must have a ref_id in struct